src/app/form-entry/control-group.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from '@angular/core';
+import { FormControl, FormGroup, Validators, FormBuilder, FormArray } from '@angular/forms';
+import { Validation } from './models/validation';
+import { QuestionBase } from './models/question-base';
+import { QuestionGroup } from './models/question-group';
+@Injectable()
+export class ControlGroupService {
+  repeating = [];
+  matches = [];
+  formValue: any;
+  controls = [];
+  constructor(private formBuilder: FormBuilder) { }
+  create(questions: QuestionBase<any>[], formKey?): any {
+    let temp = {},
+      toReturn = {},
+      self = this;
+
+    for (let question of questions) {
+      let val = question.value || '',
+        validators = null;
+
+      if (question.validation) {
+
+        if (Array.isArray(question.validation)) {
+          validators = [];
+          question.validation.forEach(i => validators.push(setValidator(i, question)));
+        } else {
+          validators = setValidator(question.validation);
+        };
+      }
+
+
+      if (question.type === 'group') {

```

Comment:
This function will potentially get too long. Consider creating functions to help you create each control type. See how angular's FormBuilder cleverly creates functions to return form control, form array and form group.